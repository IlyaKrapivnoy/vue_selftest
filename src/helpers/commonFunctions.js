import { ref } from "vue";

export const activeIndex = ref(localStorage.getItem("activeIndex") || "1");
export const updateActiveIndex = (index) => {
  activeIndex.value = index;
  localStorage.setItem("activeIndex", index);
};

export const getRandomItem = (array) => {
  if (!Array.isArray(array) || array.length === 0) return;
  const randomIndex = Math.floor(Math.random() * array.length);

  return array[randomIndex];
};
