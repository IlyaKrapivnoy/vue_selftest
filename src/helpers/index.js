import { ref } from "vue";

export const activeIndex = ref(localStorage.getItem("activeIndex") || "1");
export const setActiveIndex = (index) => {
  activeIndex.value = index;
  localStorage.setItem("activeIndex", index);
};

export function getRandomItem(array) {
  return array[Math.floor(Math.random() * array.length)];
}
