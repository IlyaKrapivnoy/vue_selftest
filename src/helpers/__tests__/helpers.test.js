import { getRandomItem } from "@/helpers/commonFunctions";

describe("getRandomItem", () => {
  it("returns a random item from the array", () => {
    const array = [1, 2, 3, 4, 5];
    const randomItem = getRandomItem(array);

    expect(array).toContain(randomItem);
  });

  it("returns undefined for an empty array", () => {
    const emptyArray = [];
    const randomItem = getRandomItem(emptyArray);

    expect(randomItem).toBeUndefined();
  });

  it("matches snapshot for random item and undefined return", () => {
    const array = [1, 2, 3, 4, 5];
    const emptyArray = [];
    const randomItemFromArray = getRandomItem(array);
    const randomItemFromEmptyArray = getRandomItem(emptyArray);

    expect(randomItemFromArray).toMatchSnapshot("randomItemFromArray");
    expect(randomItemFromEmptyArray).toMatchSnapshot(
      "randomItemFromEmptyArray"
    );
  });
});
