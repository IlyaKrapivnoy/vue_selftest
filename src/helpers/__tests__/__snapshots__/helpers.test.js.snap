// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRandomItem matches snapshot for random item and undefined return: randomItemFromArray 1`] = `1`;

exports[`getRandomItem matches snapshot for random item and undefined return: randomItemFromEmptyArray 1`] = `undefined`;
